driver/src/main/scala/api/collections/ChangeStreamOps.scala

```diff
@@ -0,0 +1,85 @@
+package reactivemongo.api.collections
+
+import scala.language.higherKinds
+
+import api.ChangeStreams
+import reactivemongo.api._
+
+trait ChangeStreamOps[P <: SerializationPack with Singleton] { collection: GenericCollection[P] =>
+
+  import BatchCommands.AggregationFramework.ChangeStream
+  // I wasn't able to prove that it is equal to `collection.pack.Value`
```

Comment:
I admit I lost the battle with the dependent types. :(

Within PipelineOperators, we only have access to types path-dependent on `AggregationPipeline.pack` (such as `AggregationPipeline.pack.Value` into which `PipelineOperator`s are serialized).

Normally, `collection.pack.type =:= collection.BatchCommands.AggregationFramework .pack.type`, but I was unable to prove it. So I used the latter here to type some arguments to the watch method.

From the user's perspective it will have no practical difference, as for any concrete instance of GenericCollection the two above types will be the same.

I can remove the comment, though... :)